Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -226,6 +226,21 @@ bool ES::LaunchPPCTitle(u64 title_id, bool skip_reload)
   s_title_context.Update(content_loader);
   INFO_LOG(IOS_ES, "LaunchPPCTitle: Title context changed: %016" PRIx64,
            s_title_context.tmd.GetTitleId());
+
+  // Note: the UID/GID is also updated for IOS titles, but since we have no guarantee IOS titles
+  // are installed, we can only do this for PPC titles.
+  IOS::ES::UIDSys uid_map{Common::FromWhichRoot::FROM_SESSION_ROOT};
```

Comment:
This seems kind of duplicated with the below code in DIVerify. This could probably be moved to its own function like:

```cpp
bool GiveThisAnAppropriateName(u64 title_id, u16 group_id)
{
  IOS::ES::UIDSys uid_sys{Common::FromWhichRoot::FROM_SESSION_ROOT};
  const u32 uid = uid_sys.AddTitle(title_id);
  if (!uid)
  {
    ERROR_LOG(IOS_ES, "Failed to get UID for title %016" PRIx64, title_id);
    return false;
  }
  SetUIDForPPC(uid);
  SetGIDForPPC(group_id);
  return true;
}
```

Then the below failure case is just:

```cpp
if (!GiveThisAnAppropriateName(title_id, s_title_context.tmd.GetGroupId()))
{
  s_title_context.Clear();
  INFO_LOG(IOS_ES, "LaunchPPCTitle: Title context changed: (none)");
  return false;
}
...
```

and the other one:

```cpp
if (!GiveThisAnAppropriateName(tmd.GetTitleId(), tmd.GetGroupId()))
{
  return ES_SHORT_READ;
}
```